scripts/report-bundle-sizes-as-a-pr-comment.js

```diff
@@ -0,0 +1,179 @@
+      {
+        method: 'GET',
+        headers: {
+          'Content-Type': 'application/json',
+          'DD-API-KEY': getOrg2ApiKey(),
+          'DD-APPLICATION-KEY': getOrg2AppKey(),
+        },
+      }
+    )
+    if (!response.ok) {
+      throw new Error(`HTTP Error Response: ${response.status} ${response.statusText}`)
+    }
+    const data = await response.json()
+    if (data.series && data.series.length > 0 && data.series[0].pointlist && data.series[0].pointlist.length > 0) {
+      return {
+        name: budget.name,
+        size: data.series[0].pointlist[0][1],
+      }
+    }
+    await timeout(TIMEOUT_DURATION_MS)
+  }
+  return {
+    name: budget.name,
+    size: null,
+  }
+}
+
+function compare(resultsBaseQuery, resultsLocalQuery) {
+  return resultsBaseQuery.map((baseResult, index) => {
+    const localResult = resultsLocalQuery[index]
+    let percentageChange = null
+
+    if (baseResult.size && localResult.size) {
+      percentageChange = (((localResult.size - baseResult.size) / baseResult.size) * 100).toFixed(2)
+    }
+    return {
+      name: baseResult.name,
+      percentageChange,
+    }
+  })
+}
+
+async function retrieveExistingCommentId(prNumber) {
+  const response = await fetch(`https://api.github.com/repos/DataDog/browser-sdk/issues/${prNumber}/comments`, {
+    method: 'GET',
+    headers: {
+      Authorization: `token ${GITHUB_TOKEN}`,
+    },
+  })
+  if (!response.ok) {
+    throw new Error(`HTTP Error Response: ${response.status} ${response.statusText}`)
+  }
+  const comments = await response.json()
+  const targetComment = comments.filter((comment) => comment.body.startsWith('## Bundles Sizes Evolution'))[0]
+  if (targetComment !== undefined) {
+    return targetComment.id
+  }
+}
+async function updateOrAddComment(difference, resultsBaseQuery, resultsLocalQuery, prNumber, commentId) {
+  const message = createMessage(difference, resultsBaseQuery, resultsLocalQuery)
+  const method = commentId ? 'PATCH' : 'POST'
+  const payload = {
+    pr_url: `https://github.com/DataDog/browser-sdk/pull/${prNumber}`,
+    message,
+    header: PR_COMMENT_HEADER,
+    org: 'DataDog',
+    repo: 'browser-sdk',
+  }
+  const response = await fetch('https://pr-commenter.us1.ddbuild.io/internal/cit/pr-comment', {
+    method,
+    headers: {
+      Authorization: `Bearer ${PR_COMMENTER_AUTH_TOKEN}`,
+    },
+    body: JSON.stringify(payload),
+  })
+  if (!response.ok) {
+    throw new Error(`HTTP Error Response: ${response.status} ${response.statusText}`)
+  }
+}
+
+function createMessage(difference, resultsBaseQuery, resultsLocalQuery) {
+  let message = '| Bundle | Base Size | Local Size | 𝚫% |\n| --- | --- | --- | --- |\n'
+  difference.forEach((diff, index) => {
+    const baseSize = `${(resultsBaseQuery[index].size / 1024).toFixed(1)} kB`
+    const localSize = `${(resultsLocalQuery[index].size / 1024).toFixed(1)} kB`
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏What about extracting a `formatSize` function?